runtime/syntax/sh.yaml

```diff
@@ -30,12 +30,12 @@ rules:
     # Numbers
     - constant.number: "\\b[0-9]+\\b"
       # Conditionals and control flow
-    - statement: "\\b(case|do|done|elif|else|esac|exit|fi|for|function|if|in|local|read|return|select|shift|then|time|until|while)\\b"
+    - statement: "\\b(break|case|continue|do|done|elif|else|esac|eval|exec|exit|fi|for|function|getopts|if|in|local|read|return|select|shift|then|time|trap|until|while)\\b"
```

Comment:
Strictly speaking, `eval`, `exec`, `getopts` and `trap` are builtins, not keywords, so perhaps it's better to move them to the "Shell commands" group below? (The same actually applies to `exit`, `read`, `shift` and `time`.)

I'm actually  not against highlighting them as keywords, if that's more practical. But then it's not clear why e.g. `exec` and `exit` are highlighted as keywords but e.g. `export` and `set` are not.

Reply:
I wrote that most of the commands added are builtin commands so that it would be known where the commands can be looked up when using something like `man`.

I do not know why `time` would be highlighted as a keyword but I was thinking that commands like `exec` and `exit` would be highlighted as such because they are used with control flow.

I think `read` and `shift` are also highlighted as keywords because they are sometimes used in loops. I realized that I have not seen `eval` being used with control flow so I am thinking about moving it to "Shell commands".

I do not mind much about how commands and keywords are highlighted so I do not know if moving builtin commands to "Shell commands" would be better, but I still do not think people would mind that much. I was looking at this file when checking how the commands would be highlighted: [example.sh.txt](https://github.com/zyedidia/micro/files/15164936/example.sh.txt)

There are still other programs but I was checking how the file is highlighted by default when using other programs just as a reference:
![hlcmp](https://github.com/zyedidia/micro/assets/127052329/b1afa984-3277-4ec6-a9d4-97e53596b192)

There is only 1 color used with commands and statements in Vim (top right) and all builtin commands are highlighted using a color that is different with statements in Discord (bottom left). `trap` and `getopts` were not highlighted in nano (bottom right) but `exec` and `exit` were highlighted as keywords like in this pull request. `set` was highlighted with the same color as `printf`.

(Note: I am not used with writing comments so I would like to also know if this comment is too detailed or long)

Reply:
I think now I understand what should be the criteria: "Conditionals and control flow" should include those commands that actually affect control flow, not just are often used in loops. I.e. `break`, `continue`, `return` - as I've realized, technically these are not keywords (i.e. not special syntax constructs), but they affect the control flow, i.e. they determine which command will be executed next, so they should be highlighted as "Conditionals and control flow" together with keywords.

See for example the `Flow-Control Constructs` section in https://linux.die.net/man/1/dash.

`exec`, `exit` - arguably these also can be considered as control-flow commands, as they affect execution of the entire script.

But `getopts`, `read`, `shift`, `time` and `trap` - these should be moved to "Shell commands", it seems.

Reply:
Well, `trap` might be an exception: it does affect the execution flow of the script (although not immediately), so we may leave it in "Conditionals and control flow".

Reply:
Also I'm not quite sure what to do with `local`, which is also technically a command, not a keyword.

`local`, `let`, `set`, `export` and `read` are all related to setting shell variables, so it might make sense to highlight them all the same way.

Reply:
I think `local` was in "Conditionals and control flow" because the value of specified variables where the command is run is not modified.

I think it would be better moving `getopts`, `local`, `read`, `shift` and `time` to "Shell commands" because control flow is not affected when they are run, but I will move them in this pull request.